docs: `ContractCreateTransaction` endpoint (#474)
hiero-ledger/hiero-sdk-tck: docs/test-specifications/contract-service/ContractCreateTransaction.md

**ContractCreate protobuf:**\
[https://github.com/hiero-ledger/hiero-consensus-node/blob/main/hapi/hedera-protobuf-java-api/src/main/proto/services/contract\_create.proto](https://github.com/hiero-ledger/hiero-consensus-node/blob/main/hapi/hedera-protobuf-java-api/src/main/proto/services/contract_create.proto)

**Mirror Node APIs:**

- Contract info: [https://mainnet.mirrornode.hedera.com/api/v1/docs/#/contracts/getContract](https://mainnet.mirrornode.hedera.com/api/v1/docs/#/contracts/getContract)
- Create results: [https://mainnet.mirrornode.hedera.com/api/v1/docs/#/contracts/getContractResultsByContractId](https://mainnet.mirrornode.hedera.com/api/v1/docs/#/contracts/getContractResultsByContractId)

## JSON-RPC API Endpoint Documentation

### Method Name

`createContract`

### Input Parameters

> | Parameter Name          | Type                                                    | Required/Optional | Description/Notes               |

Author: proxyAccountID -> deprecated and not enabled as far as I am aware
initBytecode -> where do we set this: https://github.com/hiero-ledger/hiero-sdk-js/blob/main/src/contract/ContractCreateTransaction.js. I see in the protobufs but I was not sure it was used https://github.com/hiero-ledger/hiero-consensus-node/blob/main/hapi/hedera-protobuf-java-api/src/main/proto/services/contract_create.proto#L94

> | Parameter Name          | Type                                                    | Required/Optional | Description/Notes               |

Author: ah ok thanks!

| ----------------------- | ------------------------------------------------------- | ----------------- |---------------------------------|

> | bytecodeFileId          | string                                                  | required          | ID of file containing contract bytecode |

Author: I guess that would be the same for gas right? All should be optional.

### Output Parameters

| Parameter Name | Type   | Description/Notes                                      |
| -------------- | ------ |--------------------------------------------------------|
| contractId     | string | The ID of the created contract (x.y.z)                 |
| status         | string | Hedera network response code from `TransactionReceipt` |

---

## Property Tests

### **Bytecode File ID**

| Test no | Name                                                                                                           | Input                             | Expected Response                                      | Implemented (Y/N) |
|---------|----------------------------------------------------------------------------------------------------------------|-----------------------------------|--------------------------------------------------------|-------------------|
| 1       | Create a contract with valid file containing bytecode                                                          | bytecodeFileId=\<VALID\_FILE\_ID> | Transaction succeeds, contract exists via `ContractInfoQuery`. | N                 |

> | 2       | Create a contract with missing `bytecodeFileId`                                                                | —                                 | Fails with `INVALID_FILE_ID` from network.             | N                 |

Reviewer: ```suggestion
| 2       | Create a contract with missing `bytecodeFileId` and `bytecode`                                                               | —                                 | Fails with `INVALID_FILE_ID` from network.             | N                 |
```

| 3       | Create a contract with invalid/nonexistent `bytecodeFileId`                                                    | bytecodeFileId="0.0.9999999"      | Fails with `INVALID_FILE_ID`.                          | N                 |
| 3       | Create a contract with a valid file ID but no content                                                          | bytecodeFileId="VALID/FILE/ID"    | Fails with `CONTRACT_FILE_EMPTY `.                     | N                 |
| 4       | Create a contract with a valid file ID but with no bytecode                                                    | bytecodeFileId="VALID/FILE/ID"    | Fails with `CONTRACT_BYTECODE_EMPTY`.                  | N                 |

> | 5       | Create a contract with bytecode file too large                                                                 | bytecodeFileId=\<OVERSIZED\_FILE> | Fails with `CONTRACT__SIZE_LIMIT_EXCEEDED`.            | N                 |

Reviewer: ```suggestion
| 5       | Create a contract with bytecode file too large                                                                 | bytecodeFileId=\<OVERSIZED\_FILE> | Fails with `CONTRACT_SIZE_LIMIT_EXCEEDED`.            | N                 |
```

| 8       | Create and deploy a valid  ERC-721 contract, then call one of its functions                                    | bytecodeFileId=\<VALID\_FILE>     | Succeeds with expected result                          | N                 |
| 9       | Create and deploy a valid contract that interact with the Hiero account service system contract                | bytecodeFileId=\<VALID\_FILE>     | Succeeds with correct interaction with system contract | N                 |
| 10      | Create and deploy a valid contract that interact with the Hiero exchange rate system contract                  | bytecodeFileId=\<VALID\_FILE>     | Succeeds with correct interaction with system contract | N                 |
| 11      | Create and deploy a valid contract that interact with the Hiero schedule service system contract               | bytecodeFileId=\<VALID\_FILE>     | Succeeds with correct interaction with system contract | N                 |
| 12      | Create and deploy a valid contract that interact with the Hiero token service system contract                  | bytecodeFileId=\<VALID\_FILE>     | Succeeds with correct interaction with system contract | N                 |

> | 13      | Create and deploy a valid contract that interact with the Hiero psuedo random number generator system contract | bytecodeFileId=\<VALID\_FILE>     | Succeeds with correct interaction with system contract | N                 |

Author: ok should these be testing just the creation of a contract that uses a system contract instead of creating and calling a function?

> | 13      | Create and deploy a valid contract that interact with the Hiero psuedo random number generator system contract | bytecodeFileId=\<VALID\_FILE>     | Succeeds with correct interaction with system contract | N                 |

Reviewer: Yes I think this makes most sense.

| 15      | Create and deploy a valid contract and set the file ID to be a system file under 0.0.1000                      | bytecodeFileId=\<VALID\_FILE>     | Fails                                              | N                 |
| 16      | Create and deploy a valid contract and set the file ID to be a deleted file ID                                 | bytecodeFileId=\<VALID\_FILE>     | Fails                                              | N                 |

#### JSON Example

```json
{ "method":"createContract", "params": { "bytecodeFileId":"0.0.123" } }
```

---

### **Admin Key**

| Test no | Name                                                          | Input                                                         | Expected Response                                             | Implemented |
|---------|---------------------------------------------------------------|---------------------------------------------------------------|---------------------------------------------------------------| ----------- |

> | 1       | Create a contract with a valid ED25519                        | adminKey=\<VALID\_KEY>                                        | Transaction succeeds, adminKey reflected in `getContract`.    | N           |

Reviewer: Not sure what the `\`symbols are for.

| 4       | Create a contract with invalid/nonexistent `bytecodeFileId`                                                    | bytecodeFileId="0.0.9999999"      | Fails with `INVALID_FILE_ID`.                          | N                 |
| 5       | Create a contract with a valid file ID but no content                                                          | bytecodeFileId="VALID_FILE_ID"    | Fails with `CONTRACT_FILE_EMPTY `.                     | N                 |
| 6       | Create a contract with a valid file ID but with no bytecode                                                    | bytecodeFileId="VALID_FILE_ID"    | Fails with `CONTRACT_BYTECODE_EMPTY`.                  | N                 |
| 7       | Create a contract with bytecode file too large                                                                 | bytecodeFileId="OVERSIZE_FILE"    | Fails with `CONTRACT__SIZE_LIMIT_EXCEEDED`.            | N                 |
| 8       | Create and deploy a valid contract, then call one of its functions                                             | bytecodeFileId="VALID_FILE_ID"    | Succeeds with expected result                          | N                 |

> | 9       | Create and deploy a valid  ERC-20 contract, then call one of its functions                                     | bytecodeFileId="VALID_FILE_ID"    | Succeeds with expected result                          | N                 |

Reviewer: 8 and 9 have contract execute functionality, we can keep these for the following transaction docs

| 10       | Create and deploy a valid  ERC-721 contract, then call one of its functions                                    | bytecodeFileId="VALID_FILE_ID"    | Succeeds with expected result                          | N                 |

> | 11       | Create and deploy a valid contract that uses the Hiero account service system contract                         | bytecodeFileId="VALID_FILE_ID"    | Succeeds with correct interaction with system contract | N                 |

Reviewer: Do you mean to interact (call) the system contracts or only to deploy them?
If we want to call them it should be in ContractExecute and ContractCallQuery endpoints

Author: yes needed to update those thanks!

### **Initcode**
- Initcode represents the raw bytes of the smart contract creation logic that are directly included in the transaction. 
  
| Test no | Name                                                                                  | Input                                                                              | Expected Response                                                                                                     | Implemented (Y/N) |
|---------|---------------------------------------------------------------------------------------|------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------|-------------------|
| 1       | Create a contract with valid initcode under the transaction size limit                | `initcode=<VALID_INITCODE_HEX>`, no `bytecodeFileId`                               | Transaction succeeds; contract is created and bytecode recorded in state.                                             | N                 |

> | 2       | Create a contract with initcode exactly at the transaction size limit boundary        | `initcode=<MAX_ALLOWED_HEX_BYTES>` such that total tx size == network limit        | Transaction succeeds.                                                                                                 | N                 |

Reviewer: We don't have such status `MAX_ALLOWED_HEX_BYTES`. In reality the consensus node would return grpc error which will make the sdks retry and after max attempts the transaction would fail. I suggest to not have such scenarios in the TCK.

Author: That was not supposed to be a transaction response status rather the max that would be passed in the initcode field of the transaction. Should it be called something else?

Reviewer: I read this test (test No2) wrong, we can keep it as it is. My comment should apply for test No3.

| 9       | Create a contract with an ECDSA private key as the admin key        | ECDSA adminKey= private key                                                                                                   | Transaction fails                                             | N           |
| 10      | Create a contract with valid KeyList of ED25519 and ECDSAsecp256k1 private and public keys as its admin key        | adminKey=<VALID_KEYLIST>, commonTransactionParams.signers=[<KEYS_IN_KEYLIST>]  | Transaction suceeds                                             | N           |
| 11      | Create a contract with a valid ThresholdKey of ED25519 and ECDSAsecp256k1 private and public keys as its admin key | adminKey=<VALID_THRESHOLD_KEY>, commonTransactionParams.signers=[<KEYS_IN_THRESHOLD_KEY>]| Transaction succeeds                            | N           |
| 12      | Create a contract with a valid key as the admin key but do not sign with it      | adminKey=<VALID_KEY>                                                                                             | The token creation fails with an INVALID_SIGNATURE response code from the network.         | N           |

---

> ### **Gas**

Reviewer: the gas is `int64` so we could reuse tests such as `9,223,372,036,854,775,807 (`int64` min)` ,`9,223,372,036,854,775,807 (`int64` min + 1)` etc

Reviewer: For example `AccountAllowanceApproveTransaction` suite -> `describe("ApproveHbarAllowance", function ()...`

| 3      | Create contract with admin key and zero gas                   | adminKey=valid, gas="0"                | Fails with `INSUFFICIENT_GAS`.           | N           |
| 4      | Create contract without admin key and zero gas                | adminKey=empty, gas="0"                | Fails with `INSUFFICIENT_GAS`.           | N           |
| 5      | Create contract with admin key and excessive gas over limit   | adminKey=valid, gas="9223372036854775807" | Fails with `MAX_GAS_LIMIT_EXCEEDED`.     | N           |
| 6      | Create contract without admin key and excessive gas over limit| adminKey=empty, gas="9223372036854775807" | Fails with `MAX_GAS_LIMIT_EXCEEDED`.     | N           |
| 7      | Create contract with admin key and negative gas               | adminKey=valid, gas="-1"               | Fails with `CONTRACT_NEGATIVE_GAS`.      | N           |
| 8      | Create contract without admin key and negative gas            | adminKey=empty, gas="-1"               | Fails with `CONTRACT_NEGATIVE_GAS`.      | N           |

---

> ### **Initial Balance**

Reviewer: same as `gas`

Author: I see we did not do them for accounts. We should add them here as well? https://github.com/hiero-ledger/hiero-sdk-tck/blob/add-test-specs-contract-create/docs/test-specifications/crypto-service/AccountCreateTransaction.md#initial-balance

Author: I guess balances have a limit of 50 billion for hbar since the max hbar in 50B. 

From that POV I wrote that the max tests would fail because the payer from where the balance is being transferred from would never have enough.

- A flag indicating that this smart contract declines to receive any reward for staking its HBAR balance to help secure the network.
  
| Test no | Name                                                                 | Input                      | Expected Response                               | Implemented |
|---------|----------------------------------------------------------------------| -------------------------- | ----------------------------------------------- | ----------- |
| 1       | Create a contract with an admin key that decline staking rewards     | declineStakingReward=true  | Succeeds; contract staking reward declined.     | N           |
| 2       | Create a contract with no admin key that decline staking rewards     | declineStakingReward=true  | Succeeds; contract staking reward declined.     | N           |
| 3       | Create a contarct with an admin key that that accept staking rewards | declineStakingReward=false | Succeeds; contract eligible for staking reward. | N           |
| 4       | Create a contarct with no admin key that accept staking rewards      | declineStakingReward=false | Succeeds; contract eligible for staking reward. | N           |

---

> ### **Max Automatic Associations**

Reviewer: Max automatic associations is `int32` so we can test it's boundaries as well

Author: I updated the tests with the assumption that the max limit for associations is 1,000. I will double check if this is still true

Author: actually never mind I see this HIP which removed the limitation: https://hips.hedera.com/hip/hip-367. Will update the tests accordingly

Author: i see we have these tests: https://github.com/hiero-ledger/hiero-sdk-tck/blob/add-test-specs-contract-create/docs/test-specifications/crypto-service/AccountCreateTransaction.md#max-automatic-token-associations. Seems like they also need to be updated

Reviewer: Now I remember that this HIP was not enabled for contracts.
Anyways, we can still test the type boundaries and observe the response from the consensus node.
For example if we pass `Max Int32` will observe `REQUESTED_NUM_AUTOMATIC_ASSOCIATIONS_EXCEEDS_ASSOCIATION_LIMIT` response code

Author: ok great as a follow up we should account create and update for these tests

| 4       | Create a contract with an invalid initcode hex string                        | initcode="0xZZ"                                                              | Fails with an SDK internal error.                                                                                 | N                 |
| 5       | Create a contract with a valid initcode with constructorParameters           | initcode=<VALID_INITCODE_HEX>, constructorParameters=<VALID_ABI_ENCODED_HEX> | Transaction succeeds; constructor runs with provided params (verify via `ContractFunctionResult` on mirror node). | N                 |
| 6       | Create a contract with a valid initcode but insufficient gas                 | initcode=<VALID_INITCODE_HEX>, gas="0"                                       | Transaction fails with `INSUFFICIENT_GAS`.                                                                        | N                 |

> | 7       | Create a contract with no bytecode                                           | bytecodeFileId=<VALID_FILE_ID>                                               | Fails with `CONTRACT_BYTECODE_EMPTY`.                                                                             | N                 |

Reviewer: ```suggestion
| 7       | Create a contract with empty bytecode                                           | bytecodeFileId=<VALID_FILE_ID>                                               | Fails with `CONTRACT_BYTECODE_EMPTY`.                                                                             | N                 |
```

| 1       | Create a contract with a valid ED25519 public key as its admin key                                                 | adminKey=<VALID_ED25519_PUBLIC_KEY>, commonTransactionParams.signers=[<CORRESPONDING_VALID_ED25519_PRIVATE_KEY>] | Transaction succeeds, adminKey reflected in `getContract`.                         | N           |
| 2       | Create a contract with a valid ECDSA key as its admin key                                                          | adminKey=<VALID_ECDSA_PUBLIC_KEY>, commonTransactionParams.signers=[<CORRESPONDING_VALID_ECDSA_PRIVATE_KEY>]     | Transaction succeeds, adminKey reflected in `getContract`.                         | N           |
| 3       | Create a contract wtith an Invalid key format                                                                      | adminKey=<INVALID_PUBLIC_KEY>                                                                                    | Fails with SDK error.                                                              | N           |

> | 4       | Create a contract with no adminKey                                                                                 | initcode=<VALID_INITCODE_HEX>                                                                                    | Transaction succeeds, no admin key set. Contract is immutable                      | N           |

Reviewer: ```suggestion
| 4       | Create a contract with no adminKey                                                                                 | initcode=<VALID_INITCODE_HEX>                                                                                    | Transaction succeeds, the admin key is the to the contractID of the created contract. Contract is immutable                      | N           |
```

| 5       | Create a contract with an admin key too large                                                                      | adminKey=<OVERSIZED_KEY>                                                                                         | Fails with an SDK internal error.                                                  | N           |
| 6       | Create a contract with an ED25519 complex admin key structure                                                      | ED25519 adminKey= 13/26 each with 3 keys in its sub threshold                                                    | Transaction suceeds with SUCCESS                                                   | N           |
| 7       | Create a contract with an ECDSA complex admin key structure                                                        | ECDSA adminKey= 13/26 each with 3 keys in its sub threshold                                                      | Transaction suceeds with SUCCESS                                                   | N           |

> | 8       | Create a contract with an ED25519 private key as the admin key                                                     | ED25519 adminKey=<VALID_ED25519_PRIVATE_KEY>                                                                     | Transaction fails                                                                  | N           |

Reviewer: does not fail

---

## Property Tests

### **Bytecode File ID**
- The file ID that contains the bytecode for the smart contract. The bytecode must be deployed to a file via FileCreateTransaction and the resulting file ID would be used as a valid file ID.

| Test no | Name                                                                                                           | Input                             | Expected Response                                      | Implemented (Y/N) |
|---------|----------------------------------------------------------------------------------------------------------------|-----------------------------------|--------------------------------------------------------|-------------------|
| 1       | Create a contract with valid file containing bytecode                                                          | bytecodeFileId=<VALID_FILE_ID>    | Transaction succeeds, contract exists via `ContractInfoQuery`. | N         |
| 2       | Create a contract with missing `bytecodeFileId` and `bytecode`                                                 | —                                 | Fails with `INVALID_FILE_ID` from network.             | N                 |

> | 3       | Create a contract with `bytecodeFileId` and incorrect `bytecode`                                               | bytecodeFileId=<VALID_FILE_ID>    | Fails with `INVALID_FILE_ID` from network.             | N                 |

Reviewer: added example

| 17      | Create and deploy a valid contract and set the file ID to be a deleted file ID                                 | bytecodeFileId=<VALID_FILE_ID>    | Fails with `INVALID_FILE_ID`                                                  | N                 |

### **Initcode**
- Initcode represents the raw bytes of the smart contract creation logic that are directly included in the transaction. 
  
| Test no | Name                                                                   | Input                         | Expected Response                                                         | Implemented (Y/N) |
|---------|------------------------------------------------------------------------|-------------------------------|---------------------------------------------------------------------------|-------------------|
| 1       | Create a contract with valid initcode under the transaction size limit | initcode=<VALID_INITCODE_HEX> | Transaction succeeds; contract is created and bytecode recorded in state. | N                 |

> | 2       | Create a contract with initcode exactly at the transaction size limit boundary | initcode=<MAX_ALLOWED_HEX_BYTES>` such that total tx size == network limit      | Transaction succeeds.                                                                                             | N                 |

Reviewer: flaky and invalid test scenario, since different grpc parsers can create different length for messages

> | 3       | Create a contract with missing initcode AND missing bytecodeFileId                    | No `initcode`, no `bytecodeFileId`                                                 | Transaction fails with `CONTRACT_BYTECODE_EMPTY` (or `INVALID_TRANSACTION_BODY`).                                     | N                 |

Reviewer 2: ```suggestion
| 3       | Create a contract with missing initcode AND missing bytecodeFileId                    |                                                 | Transaction fails with `CONTRACT_BYTECODE_EMPTY` (or `INVALID_TRANSACTION_BODY`).                                     | N                 |
```
Same here

> | 4       | Create a contract with both initcode and bytecodeFileId supplied                      | `initcode=<VALID_INITCODE_HEX>`, `bytecodeFileId=0.0.1234`                         | Transaction fails with `INVALID_TRANSACTION_BODY` (or SDK rejects before submission).                                 | N                 |

Reviewer 2: _Should_ the SDKs not allow this transaction to be submitted? Seems like an area we'd want to keep consistent across SDKs

Reviewer: should succeed, since sdks internally set the other property to null when setting

> | 5       | Create a contract with an invalid initcode hex string                                 | `initcode="0xZZ"`                                                                  | SDK/client validation error, transaction not submitted OR network fails with `INVALID_CONTRACT_INITCODE`.             | N                 |

Reviewer 2: Same here, we should specify exactly what's expected and not an "or"

|---------|---------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------| ----------- |
| 1       | Create a contract with a valid ED25519 public key as its admin key  | adminKey=<VALID_ED25519_PUBLIC_KEY>, commonTransactionParams.signers=[<CORRESPONDING_VALID_ED25519_PRIVATE_KEY>]              | Transaction succeeds, adminKey reflected in `getContract`.    | N           |
| 2       | Create a contract with a valid ECDSA key as its admin key           | adminKey=<VALID_ECDSA_PUBLIC_KEY>, commonTransactionParams.signers=[<CORRESPONDING_VALID_ECDSA_PRIVATE_KEY>]                  | Transaction succeeds, adminKey reflected in `getContract`.    | N           |

> | 3       | Create a contract wtith an Invalid key format                       | adminKey=<INVALID_PUBLIC_KEY>                                                                                                 | Fails with SDK error or `INVALID_ADMIN_KEY`.                                                                                  | N           |

Reviewer 2: ```suggestion
| 3       | Create a contract wtith an Invalid key format                       | adminKey=<INVALID_PUBLIC_KEY>                                                                                                 | Fails with SDK error                                                                                  | N           |
```
Invalid keys shouldn't be able to be parsed by SDKs so this should fail there

> | 4       | Create a contract with no adminKey                                  | —                                                                                                                             | Transaction succeeds, no admin key set. Contract is immutable | N           |

Reviewer 2: This looks like a copy of `bytecodeFileId` test no. 2 where there are no provided inputs, but this one succeeds and that one fails.

> | 5       | Create a contract with an admin key too large                       | adminKey=<OVERSIZED_KEY>                                                                                                      | Fails with `INVALID_ADMIN_KEY` or similar.                    | N           |

Reviewer 2: Same as the invalid key test, this should fail at the SDK level

| 6       | Create a contract with an ED25519 complex admin key structure       | ED25519 adminKey= 13/26 each with 3 keys in its sub threshold                                                                 | Transaction suceeds with SUCCESS                              | N           |
| 7       | Create a contract with an ECDSA complex admin key structure         | ECDSA adminKey= 13/26 each with 3 keys in its sub threshold                                                                   | Transaction suceeds with SUCCESS                              | N           |

> | 8       | Create a contract with an ED25519 private key as the admin key      | ED25519 adminKey= private key                                                                                                 | Transaction fails                                             | N           |

Reviewer 2: ```suggestion
| 8       | Create a contract with an ED25519 private key as the admin key      | adminKey=<VALID_ED25519_PRIVATE_KEY>                                                                                                 | Transaction fails                                             | N           |
```
Should change the input for the test below this one too

| 9       | Create a contract with an ECDSA private key as the admin key        | ECDSA adminKey= private key                                                                                                   | Transaction fails                                             | N           |
| 10      | Create a contract with valid KeyList of ED25519 and ECDSAsecp256k1 private and public keys as its admin key        | adminKey=<VALID_KEYLIST>, commonTransactionParams.signers=[<KEYS_IN_KEYLIST>]  | Transaction suceeds                                             | N           |
| 11      | Create a contract with a valid ThresholdKey of ED25519 and ECDSAsecp256k1 private and public keys as its admin key | adminKey=<VALID_THRESHOLD_KEY>, commonTransactionParams.signers=[<KEYS_IN_THRESHOLD_KEY>]| Transaction succeeds                            | N           |

> | 12      | Create a contract with a valid key as the admin key but do not sign with it      | adminKey=<VALID_KEY>                                                                                             | The token creation fails with an INVALID_SIGNATURE response code from the network.         | N           |

Reviewer 2: Do all these tests need to be signed with the new admin key being set? Some are missing signer inputs (`commonTransactionParams.signer=[...]`)

Reviewer: done

| 10     | Create contract with admin key and gas 9,223,372,036,854,775,806 (int64 max - 1)  |adminKey=valid, gas="9223372036854775806"| The transaction succeeds, contract deployed | N   |
| 11     | Create contract with admin key and gas 9,223,372,036,854,775,807 (int64 max)  |adminKey=valid, gas="9223372036854775807"| The transaction succeeds, contract deployed | N   |
| 12     | Create contract without admin key and gas 9,223,372,036,854,775,807 (int64 max)  |adminKey=empty, gas="9223372036854775807"| The transaction succeeds, contract deployed | N   |
| 13     | Create contract with admin key and gas -9,223,372,036,854,775,808 (int64 min)  |adminKey=valid, gas="-9,223,372,036,854,775,808"| The transaction fails | N   |
| 14     | Create contract without admin key and gas -9,223,372,036,854,775,808 (int64 min)  |adminKey=empty, gas="-9,223,372,036,854,775,808"| The transaction fails| N |
| 15     | Create contract with admin key and gas -9,223,372,036,854,775,807 (int64 min + 1)  |adminKey=valid, gas="-9,223,372,036,854,775,807"| The transaction fails| N   |

> | 16     | Create contract without admin key and gas -9,223,372,036,854,775,807 (int64 min + 1)  |adminKey=empty, gas="-9,223,372,036,854,775,807"| The transaction fails | N |

Reviewer 2: The tests with `adminKey=empty` can just not provide an `adminKey` input. Also `adminKey=valid` --> `adminKey=<VALID_ADMIN_KEY>` to stay consistent

Reviewer 2: This is the same for `Constructor Parameters` tests and `Max Automatic Associations`.

| 10      | Create contract without admin key and initial balance = int64 min    | initialBalance="-9223372036854775808"     | Fails with `CONTRACT_NEGATIVE_VALUE                  | N           |
| 11      | Create contract with admin key and initial balance = int64 min + 1   | initialBalance="-9223372036854775807"     | Fails with `CONTRACT_NEGATIVE_VALUE                  | N           |
| 12      | Create contract without admin key and initial balance = int64 min + 1| initialBalance="-9223372036854775807"     | Fails with `CONTRACT_NEGATIVE_VALUE                  | N           |
| 13       | Create contract with admin key and initial balance = int64 max - 1   | initialBalance="9223372036854775806"      | Fails with `INSUFFICIENT_PAYER_BALANCE`.            | N           |
| 14      | Create contract without admin key and initial balance = int64 max - 1| initialBalance="9223372036854775806"      | Fails with `INSUFFICIENT_PAYER_BALANCE`.             | N           |
| 15      | Create contract with admin key and initial balance = int64 max       | initialBalance="9223372036854775807"      | Fails with `INSUFFICIENT_PAYER_BALANCE`.             | N           |

> | 16      | Create contract without admin key and initial balance = int64 max    | initialBalance="9223372036854775807"      | Fails with `INSUFFICIENT_PAYER_BALANCE`.             | N           |

Reviewer 2: Tests that create a contract with an admin key don't provide an admin key input

Reviewer 2: This is the same for `Auto-Renew Period`, `Memo`, `Staked Account/Node ID`, and `Decline Staking Rewards`.

| 2       | Create a contract that does not have an admin key and staked to valid node ID                                         | stakedNodeId="<VALID_NODE_ID>"                                                                                          | Succeeds, staking reflected.          | N           |
| 3       | Create a contract that does not have an admin key and that has an invalid stakedAccountId                             | stakedAccountId="0.0.99999"                                                                                             | Fails with `INVALID_STAKING_ID`.      | N           |
| 4       | Create a contract that does not have an admin key and has an invalid stakedNodeId                                     | stakedNodeId="9999999"                                                                                                  | Fails with `INVALID_STAKING_ID`.      | N           |

> | 5       | Create a contract that does not have an admin key and that tries to set both stakedAccountId and stakedNodeId present | both fields set                                                                                                         | Fails with `INVALID_STAKING_ID`.      | N           |

Reviewer: succeeds

| 7       | Create a contract that does have an admin key and staked to valid node ID                                             | adminKey=<VALID_ADMIN_KEY>, stakedNodeId="<VALID_NODE_ID>", commonTransactionParams.signers=[<VALID_ADMIN_KEY>]         | Succeeds, staking reflected.          | N           |
| 8       | Create a contract that does have an admin key and that has an invalid stakedAccountId                                 | adminKey=<VALID_ADMIN_KEY>, stakedAccountId="0.0.99999", commonTransactionParams.signers=[<VALID_ADMIN_KEY>]            | Fails with `INVALID_STAKING_ID`.      | N           |
| 9       | Create a contract that does have an admin key and has an invalid stakedNodeId                                         | adminKey=<VALID_ADMIN_KEY>, stakedNodeId="9999999", commonTransactionParams.signers=[<VALID_ADMIN_KEY>]                 | Fails with `INVALID_STAKING_ID`.      | N           |

> | 10      | Create a contract that does have an admin key and that tries to set both stakedAccountId and stakedNodeId present     | adminKey=<VALID_ADMIN_KEY>, both fields set, commonTransactionParams.signers=[<VALID_ADMIN_KEY>]                        | Fails with `INVALID_STAKING_ID`.      | N           |

Reviewer: succeeds
